src/marketing/buttons/button.scss

```diff
@@ -1,117 +1,92 @@
+    &::before {
+      opacity: 1;
+      transition: opacity 0.4s;
+    }
   }
 }
 
 .btn-primary-mktg {
-  color: var(--color-mktg-btn-primary-text);
-  background-color: var(--color-mktg-btn-primary-bg);
-  border-color: var(--color-mktg-btn-primary-border);
-
-  &:hover {
-    background-color: var(--color-mktg-btn-primary-hover-bg);
-    border-color: var(--color-mktg-btn-primary-hover-border);
-  }
-
-  &:focus {
-    box-shadow: var(--color-mktg-btn-primary-focus-shadow);
-  }
+  @include btn-solid-mktg(
+    var(--color-text-white),
+    lighten($primary-green-mktg, 5%),
+    $primary-green-mktg,
+    lighten($secondary-green-mktg, 5%),
+    $secondary-green-mktg
+  );
+}
 
-  &:disabled,
-  &.disabled,
-  &[aria-disabled=true] {
-    color: var(--color-mktg-btn-primary-disabled-text);
-    background-color: var(--color-mktg-btn-primary-disabled-bg);
-    border-color: var(--color-mktg-btn-primary-disabled-border);
-  }
+.btn-purple-mktg {
+  @include btn-solid-mktg(
+    var(--color-text-white),
+    lighten($primary-purple-mktg, 5%),
+    $primary-purple-mktg,
+    lighten($secondary-purple-mktg, 5%),
+    $secondary-purple-mktg
+  );
 }
 
 .btn-outline-mktg {
-  color: var(--color-mktg-btn-outline-text);
-  background-color: var(--color-mktg-btn-outline-bg);
-  border-color: var(--color-mktg-btn-outline-border);
-
-  &:hover {
-    color: var(--color-mktg-btn-outline-hover-text);
-    background-color: var(--color-mktg-btn-outline-hover-bg);
-    border-color: var(--color-mktg-btn-outline-hover-border);
-  }
-
-  &:focus {
-    box-shadow: var(--color-mktg-btn-outline-focus-shadow);
-  }
-
-  &:disabled,
-  &.disabled,
-  &[aria-disabled=true] {
-    color: var(--color-mktg-btn-outline-disabled-text);
-    background-color: var(--color-mktg-btn-outline-disabled-bg);
-    border-color: var(--color-mktg-btn-outline-disabled-border);
-  }
+  @include btn-outline-mktg($primary-blue-mktg, $secondary-blue-mktg, $primary-blue-mktg);
+  transition: box-shadow 0.4s, color 0.4s;
 }
 
 .btn-transparent {
-  color: var(--color-mktg-btn-dark-text);
-  background-color: var(--color-mktg-btn-dark-bg);
-  border-color: var(--color-mktg-btn-dark-border);
-
-  &:hover {
-    color: var(--color-mktg-btn-dark-hover-text);
-    background-color: var(--color-mktg-btn-dark-hover-bg);
-    border-color: var(--color-mktg-btn-dark-hover-border);
-  }
+  @include btn-outline-mktg($static-color-white, $static-color-white, $static-color-white);
+  transition: box-shadow 0.4s, color 0.4s;
+}
 
-  &:focus {
-    box-shadow: var(--color-mktg-btn-dark-focus-shadow);
-  }
+// Size modifiers
 
-  &:disabled,
-  &.disabled,
-  &[aria-disabled=true] {
-    color: var(--color-mktg-btn-dark-disabled-text);
-    background-color: var(--color-mktg-btn-dark-disabled-bg);
-    border-color: var(--color-mktg-btn-dark-disabled-border);
-  }
+.btn-sm-mktg {
+  // stylelint-disable-next-line primer/spacing
+  padding: rem(10px) rem($spacer-4) rem(12px);
```

Comment:
`padding: rem(10px) rem($spacer-4) rem(13px);`

I bumped up the bottom padding here slightly for when it uses Alliance so the padding is optically correct.

Reply:
👍  Primer CSS uses the "system" font, so might wanna have the same top/bottom padding so it doesn't look mis-aligned? And maybe also a less bold `font-weight`?

Reply:
I tweaked the small ones with more spacing below, and also:
- Reduced spacing on the sides, to $spacer-3, for better balance
- Added one pixel of spacing to the `lg` size modifier

Since these are primarily for use with Alliance, I think it's more important that they work with that font for now. In the cases when they are not used with Alliance (I'm not sure if they should be at all), the padding can be overridden. However, I think we should consider this a bug with how Alliance is set up, and we should investigate if we can export Alliance with different settings to prevent us from "hacking" the vertical alignment in this way.

Preview:

<img width="679" alt="Screen Shot 2021-04-22 at 11 30 16" src="https://user-images.githubusercontent.com/211284/115691767-a4149e00-a35e-11eb-9569-83930e9cd6f5.png">

Reply:
> However, I think we should consider this a bug with how Alliance is set up, and we should investigate if we can export Alliance with different settings to prevent us from "hacking" the vertical alignment in this way.

Yeah, I guess when we update the [Marketing Typorgraphy](https://primer.style/css/utilities/marketing-type) there is no way around to try include Alliance. Is Alliance an open source font or is a license needed to use it? [Looks like the later](https://degarism.com/Alliance). So maybe we could include it in the marketing font stack and then document that all the marketing components are optimized for Alliance. But maybe out of scope for this PR.